nservicebus/recoverability/index.md

```diff
@@ -0,0 +1,134 @@
+---
+title: Recoverability
+summary: Don't try to handle exceptions in the message handlers. Let NServiceBus do it for you.
+tags:
+- Exceptions
+- Error Handling
+- Retry
+- Recoverability
+reviewed: 2016-03-31
+redirects:
+ - nservicebus/how-do-i-handle-exceptions
+ - nservicebus/errors
+ - nservicebus/automatic-retries
+related:
+- nservicebus/operations/transactions-message-processing
+---
+Sometimes processing of a message fails. This could be due to a transient problem like a deadlock in the database, in which case retrying the message a few times should solve the issue. If the problem is more protracted, like a third party web service going down or a database being unavailable, solving the issue would take longer. It is therefore useful to wait longer before retrying the message again.
+
+NServiceBus has a built-in error handling capability called Recoverability. Recoverability enables to recover automatically or in exceptional scenarios manually from message failures. Recoverability wraps the message handling logic, including the user code with various layers of retrying logic. NServiceBus differentiates two types of retrying behaviors:
+
+* Immediate Retries (previously known as First-Level-Retries)
+* Delayed Retries (previously known as Second-Level-Retries)
+
+An oversimplified mental model for Recoverability could be thought of a try / catch block surrounding the message handling infrastructure wrapped in a for loop like the following pseudo-code
+
+```cs
```

Comment:
It's a sudo code and does not reference NServiceBus types in any way - so I'm not sure it will add any value